upload-log/action.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Upload Log'
+description: 'Upload log to AWS S3'
+inputs:
+  s3-bucket:
+    description: 'The AWS S3 bucket that stores log'
+    required: true
+  log-path:
+    description: 'The path to the log that will be uploaded'
+    default: logs
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Create tarball
+      run: tar --owner=0 --group=0 -czvf ${GITHUB_RUN_ID}.tar.gz ${{ inputs.log-path }}
+      shell: bash
+    - name: Upload to S3
+      run: aws s3api put-object --bucket ${{ inputs.s3-bucket }} --key ${GITHUB_RUN_ID}.tar.gz --body ${GITHUB_RUN_ID}.tar.gz
+      shell: bash
+    - name: Print access URL
+      env:
+        BUCKET: ${{ inputs.s3-bucket }}
+      run: |
+        echo "Access log: https://us-east-1.console.aws.amazon.com/s3/object/${BUCKET}?region=us-east-1&prefix=${GITHUB_RUN_ID}.tar.gz"
+      shell: bash
```

Comment:
Side thought:

This specifies "logs" but this could really be just a generic utility to compress a upload the contents of a directory. That made me think that maybe there is already a github action that does this?

I know we have some custom permissions on line#16 but at a minimum perhaps this is worth making more generic?

Reply:
Sure, probably we can use the word `artifact` which aligns with AWS CodeBuild.